### DISCUSSION TITLE
'define' in global javascript scope is overridden by other libraries [Bug]

### DISCUSSION THREAD

**Discussion by tgaarde** [microsoft/monaco-editor#3859] [Q&A]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [X] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Reproduction Steps

Use monaco javascript library in any project (blazor wasm)
  {
    "library": "monaco-editor@0.34.1",
    "destination": "wwwroot/lib/monaco-editor/"
  }
  
  and use a library that is using "define" as a keyword in the global javascript scope
  (Telerik.UI.for.Blazor or BlazorAppInsights)

### Actual (Problematic) Behavior

Monaco editor will not function as expected, because it's variable is overridden. 

### Expected Behavior

i expect the "define" to be either scoped away from the global scope, or renamed to something less general like "monacoDefine". 
if i download the source code and change the "var define" and all its references to "monacoDefine" it works as i expect

**Comment by supuwoerc** [comment]
+1

**Comment by zeropool** [comment]
+1
